extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesTaskRunner.java

```diff
@@ -0,0 +1,552 @@
+    return client.waitForJobCompletion(
+        k8sTaskId,
+        KubernetesTaskRunnerConfig.toMilliseconds(k8sConfig.maxTaskDuration),
+        TimeUnit.MILLISECONDS
+    );
+  }
+
+  @Override
+  public void updateStatus(Task task, TaskStatus status)
+  {
+    TaskRunnerUtils.notifyStatusChanged(listeners, task.getId(), status);
+  }
+
+  @Override
+  public void updateLocation(Task task, TaskLocation location)
+  {
+    TaskRunnerUtils.notifyLocationChanged(listeners, task.getId(), location);
+  }
+
+  @Override
+  public void shutdown(String taskid, String reason)
+  {
+    client.cleanUpJob(new K8sTaskId(taskid));
+  }
+
+
+  @Override
+  public Optional<InputStream> streamTaskReports(String taskid)
+  {
+    return Optional.absent();
+  }
+
+  @Override
+  public List<Pair<Task, ListenableFuture<TaskStatus>>> restore()
+  {
+    return ImmutableList.of();
+  }
+
+  @Override
+  public void start()
+  {
+    log.info("Starting cleanup executor for jobs older than 1 day....");
+    cleanupExecutor.scheduleAtFixedRate(
+        () ->
+            client.cleanCompletedJobsOlderThan(
+                KubernetesTaskRunnerConfig.toMilliseconds(k8sConfig.taskCleanupDelay),
+                TimeUnit.MILLISECONDS
+            ),
+        1,
+        KubernetesTaskRunnerConfig.toMilliseconds(k8sConfig.taskCleanupInterval),
+        TimeUnit.MILLISECONDS
+    );
+    log.debug("Started cleanup executor for jobs older than 1 day....");
+  }
+
+
+  @Override
+  public void stop()
+  {
+    log.info("Stopping...");
+    cleanupExecutor.shutdownNow();
+    log.info("Stopped...");
+  }
+
+  @Override
+  public Map<String, Long> getTotalTaskSlotCount()
+  {
+    return ImmutableMap.of("taskQueue", (long) taskQueueConfig.getMaxSize());
+  }
+
+  private List<String> javaOpts(Task task)
+  {
+    final List<String> javaOpts = new ArrayList<>();
+    Iterables.addAll(javaOpts, k8sConfig.javaOptsArray);
+
+    // Override task specific javaOpts
+    Object taskJavaOpts = task.getContextValue(
+        ForkingTaskRunnerConfig.JAVA_OPTS_PROPERTY
+    );
+    if (taskJavaOpts != null) {
+      Iterables.addAll(
+          javaOpts,
+          new QuotableWhiteSpaceSplitter((String) taskJavaOpts)
+      );
+    }
+
+    javaOpts.add(StringUtils.format("-Ddruid.port=%d", DruidK8sConstants.PORT));
+    javaOpts.add(StringUtils.format("-Ddruid.plaintextPort=%d", DruidK8sConstants.PORT));
+    javaOpts.add(StringUtils.format("-Ddruid.tlsPort=%d", node.isEnableTlsPort() ? DruidK8sConstants.TLS_PORT : -1));
+    javaOpts.add(StringUtils.format(
+        "-Ddruid.task.executor.tlsPort=%d",
+        node.isEnableTlsPort() ? DruidK8sConstants.TLS_PORT : -1
+    ));
+    javaOpts.add(StringUtils.format("-Ddruid.task.executor.enableTlsPort=%s", node.isEnableTlsPort())
+    );
+    return javaOpts;
+  }
+
+  private List<String> generateCommand(Task task)
+  {
+    final List<String> command = new ArrayList<>();
+    // must use deep storage shuffle now with k8s peon tasks.
```

Comment:
Do we need this comment here?

Reply:
good point, that is left-over from something.  I'll remove it.